inference/enterprise/workflows/core_steps/flow_control/condition.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Dict, List, Literal, Tuple, Type, Union
+
+from pydantic import ConfigDict, Field
+
+from inference.enterprise.workflows.core_steps.common.operators import (
+    OPERATORS_FUNCTIONS,
+    Operator,
+)
+from inference.enterprise.workflows.entities.base import OutputDefinition
+from inference.enterprise.workflows.entities.types import (
+    BOOLEAN_KIND,
+    FLOAT_KIND,
+    INTEGER_KIND,
+    LIST_OF_VALUES_KIND,
+    STRING_KIND,
+    FlowControl,
+    StepOutputSelector,
+    StepSelector,
+)
+from inference.enterprise.workflows.prototypes.block import (
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LONG_DESCRIPTION = """
+Create a branch of logic that runs only when a specified condition is met.
+
+This block is the "if statement" in Workflows.
+
+This block is responsible for flow-control in execution graph based on the condition 
+defined in its body.
+
+Right now, this block is only capable to make conditions based on output of binary 
+operators that takes two operands. 
+
+*The `Condition` block only works when a  single image is provided to the input of the 
```

Comment:
Blocks operate on batches of images/predictions, if conditions block was also operating on batches of images/predictions then it could produce output where for some elements of batch it would produce different result than for rest of the batch. If we had 3 photos in batch, 2 of them were containing cat and one was containing car, the condition would be to terminate if cat is detected - for such batch it would be unclear what to do. The logic would have to be geared towards batch (i.e. `all`, `any`, `none` on top of conditions we already have)